cmd/aws-signing-proxy/main.go

```diff
@@ -80,11 +82,23 @@ func main() {
 		AuthClient:      vaultClient,
 	})
 	listenString := fmt.Sprintf(":%v", *portFlag)
+	healthPortString := fmt.Sprintf(":%v", *healthPortFlag)
 	log.Printf("Listening on %v\n", listenString)
 	log.Printf("- Forwarding Traffic to '%s'\n", targetURL)
 	log.Printf("- Using Credentials from from Vault '%s' with credentialsPath '%s'\n", e.VaultUrl, e.VaultCredentialsPath)
 
+	go provideHealthEndpoint(healthPortString)
+
 	log.Fatal(http.ListenAndServe(listenString, signingProxy))
+
+}
+
+func provideHealthEndpoint(h string) {
+	http.HandleFunc("/status/health", func(w http.ResponseWriter, request *http.Request) {
+		w.WriteHeader(200)
+		_, _ = w.Write([]byte("ok"))
+	})
+	func() { log.Fatal(http.ListenAndServe(h, nil)) }()
```

Comment:
you don't need to wrap this call in a function here

`log.Fatal(http.ListenAndServe(h, nil))`

would be enough in this line :)